### DISCUSSION TITLE
iErrorSignatureNotFoundError: Encoded error signature {{signature}} not found on ABI. Make sure you are using the correct ABI and that the error exists on it.

### DISCUSSION THREAD

**Discussion by nk-hcode** [wevm/viem#712] [General]
When i am using wagmi with viem to mint an NFT then it is giving me this error but using ethers i am able to mint, please let me know what i need to do here to be able to mint using wagmi and viem.

**Comment by jxom** [comment]
Can you provide us with a minimal reproducible example?

**Comment by nk-hcode** [comment]
Yes, so I am using wag hooks to mint a NFT like as follows:
Hook for contract write with a function name : const { data, write } = useContractWrite({ address: contractAddresses, 
abi: contractAbi,
functionName: 'redeemForNFT', 
});

Now after this we are using write function to mint NFT like this : write?.({ args: [1], value: BigInt(nftPrice as string) });

Here sending token and price for that nft and when this function runs then I am getting error like this 



Using versions as follows: 
Wagmi : 1.2.0
Viem : 1.0.7
